File: backend/dotnet/clean-architecture/minimal/src/Core/App.Template.Application/DTOs/Auth/LoginResponseDto.cs
namespace AppTemplate.Application.DTOs.Auth;

/// <summary>
/// Login response with JWT token
/// </summary>
public record LoginResponseDto
{
    /// <summary>
    /// JWT access token
    /// </summary>
    public string AccessToken { get; init; } = string.Empty;

    /// <summary>
    /// Token expiration time in seconds
    /// </summary>
    public int ExpiresIn { get; init; }

    /// <summary>
    /// User information
    /// </summary>
    public UserInfoDto? User { get; init; }
}

File: backend/nestjs/clean-architecture/full/src/modules/auth/application/dto/login-response.dto.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { UserDto } from '@/modules/user-management/application/dto/user.dto';

export class LoginResponseDto {
  @ApiProperty({ description: 'JWT access token' })
  accessToken: string;

  @ApiProperty({ description: 'Refresh token' })
  refreshToken: string;

  @ApiProperty({ description: 'Access token expiration time in seconds' })
  expiresIn: number;

  @ApiPropertyOptional({ description: 'User information (included on login/register, omitted on refresh)' })
  user?: UserDto;

  constructor(
    accessToken: string,
    refreshToken: string,
    expiresIn: number,
    user?: UserDto,
  ) {
    this.accessToken = accessToken;
    this.refreshToken = refreshToken;
    this.expiresIn = expiresIn;
    if (user) {
      this.user = user;
    }
  }
}
